scripts/release.py

```diff
@@ -79,12 +79,19 @@ def fix_formatting():
     call(["pre-commit", "run", "--all-files"])
 
 
+def check_links():
+    """Runs sphinx-build to check links"""
+    print(f"{Fore.CYAN}[generate.check_links] {Fore.RESET}Checking links")
+    check_call(["tox", "-e", "docs-checklinks"])
+
+
 def pre_release(version):
     """Generates new docs, release announcements and creates a local tag."""
     announce(version)
     regen()
     changelog(version, write_out=True)
     fix_formatting()
+    check_links()
```

Comment:
Question to the maintainers: 
Should this be mandatory all the time or should there be the possibility to skip this with a command line flag in `release.py`?
Just thinking about a broken/flaky link could prevent an ad-hoc release of pytest (in case of a bad release that needs to be patched quickly)

Reply:
IMO link-checking should be a separate thing, where maintainers (or contributors!) can run the script but it doesn't happen in CI - flakiness is inevitable if we automate it.  It should probably go in the "how to release" docs though 👍

Reply:
In https://github.com/pytest-dev/pytest/issues/1722#issuecomment-499690461 @nicoddemus mentioned:

> @mandarvaze just call it from release.py should be fine. 👍

So I followed this. I'm up for any desired change, though.

Reply:
@Zac-HD That's already the case - from what I can see `release.py` doesn't get run on CI, but gets run when cutting a release (via `tox -e release`).

Reply:
Alright!  No change needed then :smile: 

Thanks @The-Compiler for clarifying and @steffenschroeder for your work on this :heart: